op-e2e/actions/proofs/helpers/env.go

```diff
@@ -155,28 +156,42 @@ func (env *L2FaultProofEnv) RunFaultProofProgram(t helpers.Testing, l2ClaimBlock
 		apply(fixtureInputs)
 	}
 
-	// Run the fault proof program from the state transition from L2 block 0 -> 1.
-	programCfg := NewOpProgramCfg(
-		t,
-		env,
-		fixtureInputs,
-	)
-	withInProcessPrefetcher := host.WithPrefetcher(func(ctx context.Context, logger log.Logger, kv kvstore.KV, cfg *config.Config) (host.Prefetcher, error) {
-		// Set up in-process L1 sources
-		l1Cl := env.Miner.L1Client(t, env.sd.RollupCfg)
-		l1BlobFetcher := env.Miner.BlobStore()
-
-		// Set up in-process L2 source
-		l2ClCfg := sources.L2ClientDefaultConfig(env.sd.RollupCfg, true)
-		l2RPC := env.Engine.RPCClient()
-		l2Client, err := host.NewL2Client(l2RPC, env.log, nil, &host.L2ClientConfig{L2ClientConfig: l2ClCfg, L2Head: cfg.L2Head})
-		require.NoError(t, err, "failed to create L2 client")
-		l2DebugCl := &host.L2Source{L2Client: l2Client, DebugClient: sources.NewDebugClient(l2RPC.CallContext)}
-
-		return prefetcher.NewPrefetcher(logger, l1Cl, l1BlobFetcher, l2DebugCl, kv), nil
-	})
-	err = host.FaultProofProgram(t.Ctx(), env.log, programCfg, withInProcessPrefetcher)
-	tryDumpTestFixture(t, err, t.Name(), env, programCfg)
+	// Run the fault proof program from the state transition from L2 block l2ClaimBlockNum - 1 -> l2ClaimBlockNum.
+	workDir := t.TempDir()
+	if IsKonaConfigured() {
```

Comment:
This could be cleaner, but didn't feel right to prematurely abstract over multiple program kinds. Can do tho, curious your take @ajsutton.

Reply:
Yeah I'd go with hard coding for now too. The generic solution is to export the chain data somehow rather than trying to support running every possible FPP from in the action tests.